watertap/tools/oli_api/flash.py

```diff
@@ -75,420 +73,1163 @@
 _logger.addHandler(handler)
 _logger.setLevel(logging.DEBUG)
 
-# TODO: consider config for file_name for each writing method
-
 
 class Flash:
     """
-    A class to execute OLI Cloud flash calculations, replacing and augmenting WaterAnalysis class functionality.
-
-    :param water_analysis_properties: dictionary containing pre-built water analysis input template blocks
-    :param optional_properties: dictionary containing pre-configured optional properties to attach to OLI calls (defaults to True for all properties)
-    :param input_unit_set: dictionary containing conversions between OLI and Pyomo unit names
-    :param output_unit_set: dictionary containing preferred units for output expression
-    :param stream_output_options: dictionary pointing to properties that can be extracted from flash stream outputs
-    :param relative_inflows: bool to switch between additive and absolute substitutions for solutes when using surveys
-    :param interactive_mode: bool to switch level of logging display from info to debug only
+    A class to execute OLI Cloud flash calculations.
+
+    :param optional_properties: dictionary for optional properties to attach to OLI calls, all True by default
+    :param input_unit_set: dictionary for conversions between OLI and Pyomo unit names
+    :param output_unit_set: dictionary for preferred output units
+    :param relative_inflows: bool switch for surveys - true to add specified value to initial value, false to replace initial value with specified value
+    :param interactive_mode: bool switch for level of logging display
     """
 
     def __init__(
         self,
-        water_analysis_properties=water_analysis_properties,
         optional_properties=optional_properties,
         input_unit_set=input_unit_set,
         output_unit_set=output_unit_set,
-        stream_output_options=stream_output_options,
         relative_inflows=True,
         interactive_mode=True,
     ):
-        # set values based on inputs
-        self.water_analysis_properties = water_analysis_properties
         self.optional_properties = optional_properties
         self.input_unit_set = input_unit_set
         self.output_unit_set = output_unit_set
-        self.stream_output_options = stream_output_options
         self.relative_inflows = relative_inflows
-        self.water_analysis_input_list = []
+
         if interactive_mode:
             _logger.setLevel(logging.INFO)
         else:
             _logger.setLevel(logging.DEBUG)
 
-    def build_survey(self, survey_arrays, get_oli_names=False, file_name=None):
+    def configure_water_analysis(
+        self,
+        inflows,
```

Comment:
```suggestion
        inflows=None,
```
As is, if I call this method and don't provide inflows, I will get a different exception related to `configure_water_analysis` missing a positional argument. That is, the intended `RuntimeError: Inflows must be defined for Water Analysis.`  exception, currently at line 160, would never raise.

Reply:
In fact, one question I have is why did you make this a method, and why is `inflows` required at this stage?

Reply:
I made this method to make it more user friendly to configure and save the input JSON for water analysis based on the arguments provided - `inflows` is required because it is essential to that JSON input